clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -6165,6 +6165,10 @@ def ext_initializer_string_for_char_array_too_long : ExtWarn<
 def warn_missing_field_initializers : Warning<
   "missing field %0 initializer">,
   InGroup<MissingFieldInitializers>, DefaultIgnore;
+// The same warning, but another group is needed to disable it separately.
+def warn_missing_designated_field_initializers : Warning<
+  "missing field %0 initializer">,
+  InGroup<MissingDesignatedFieldInitializers>, DefaultIgnore;
```

Comment:
```suggestion
// The same warning, but another group is needed to disable it separately.
def warn_missing_designated_field_initializers : Warning<
  warn_missing_field_initializers.Summary>,
  InGroup<MissingDesignatedFieldInitializers>, DefaultIgnore;
```
This way we don't have to worry about the two diagnostics getting different wording.